x-pack/legacy/plugins/actions/server/builtin_action_types/email.ts

```diff
@@ -173,33 +191,26 @@ async function executor(
 
 // utilities
 
-const ValidServiceNames = getValidServiceNames();
-
-function isValidService(service: string): boolean {
-  return ValidServiceNames.has(service.toLowerCase());
-}
-
-function getValidServiceNames(): Set<string> {
-  const result = new Set<string>();
+const ServiceNameHosts = getServiceNameHosts();
 
-  // add our special json service
-  result.add(JSON_TRANSPORT_SERVICE);
+// returns map of nodemailer service name: resulting host name
+function getServiceNameHosts(): Map<string, string> {
+  const result = new Map<string, string>();
 
-  const keys = Object.keys(nodemailerServices) as string[];
-  for (const key of keys) {
-    result.add(key.toLowerCase());
+  for (const [serviceName, serviceValue] of Object.entries<any>(nodemailerServices)) {
```

Comment:
nit. Is there a reason why we have to use `any` here, rather than a known type?

Reply:
I'm always hesitant to apply non-loose typing to things from the JS world.  But I poked a bit, and found this: https://github.com/nodemailer/nodemailer/blob/505c3f9ae06658ba80720f6b49a659588bb82b70/lib/well-known/index.js#L37-L47

It's slightly better than the way we are checking well-known types right now, perhaps.  It basically return `any | false`, where the `any` is very likely an object that would be fine as a `Record<string, any>`.  I think we'd still want to be pretty conservative about dereferencing the value tho.